saleor/graphql/checkout/types.py

```diff
@@ -478,3 +485,18 @@ def with_product_types(product_types):
     @staticmethod
     def resolve_language_code(root, _info, **_kwargs):
         return LanguageCodeEnum[str_to_enum(root.language_code)]
+
+    @staticmethod
+    @traced_resolver
+    def resolve_stock_reservation_expires(root: models.Checkout, info):
+        def get_oldest_stock_reservation_expiration_date(reservations):
+            if not reservations:
+                return None
+
+            return min(reservation.reserved_until for reservation in reservations)
```

Comment:
Assuming that I have two lines:
1st - expires 12:00 29-07-2020
2nd - expires 12:05 29-07-2020
Now, I am adding a new line to checkout, which gives me 
3nd - expires 22:00 29-07-2020
calling `resolve_stock_reservation_expires` will return me `12:00 29-07-2020`. Is that correct? I though that adding new line to checkout will 'refresh' reservation time (maybe it is done in different place and I just missed that)

Reply:
This was never named a requirement AFAIR. Reservations are supposed to increase your chances to complete checkout within given time, but its still on first come-first serve basis, so if you want to buy that pirate hoodie, better hurry up with picking those sneakers you've also wanted. But maybe this is worth revisiting? CC @patrys

Reply:
@patrys  What do you thin?

Reply:
How it's supposed to work is it should reset all line reservations every time you add something to your cart. In the above example, adding the third line should re-reserve all three lines. We don't want to punish people for buying multiple items.